src/test/test-firewall-util.c

```diff
@@ -1,11 +1,69 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
+#include <arpa/inet.h>
+#include <stdlib.h>
 
 #include "firewall-util.h"
 #include "log.h"
+#include "random-util.h"
 #include "tests.h"
 
 #define MAKE_IN_ADDR_UNION(a,b,c,d) (union in_addr_union) { .in.s_addr = htobe32((uint32_t) (a) << 24 | (uint32_t) (b) << 16 | (uint32_t) (c) << 8 | (uint32_t) (d))}
 
+static void make_in6_addr_union(const char *addr, union in_addr_union *u) {
+
+        if (inet_pton(AF_INET6, addr, &u->in6) < 0)
+                abort();
```

Comment:
please writ as:

```
assert_se(inet_pton() >= 0);
```

makes logging more useful, as the failing expression will be shown